extensions/blocks/send-a-message/whatsapp-button/edit.js

```diff
@@ -0,0 +1,256 @@
+/**
+ * External dependencies
+ */
+import { __, _x } from '@wordpress/i18n';
+import { useEffect, useState, useCallback } from '@wordpress/element';
+import { useDispatch } from '@wordpress/data';
+import {
+	Button,
+	BaseControl,
+	TextControl,
+	TextareaControl,
+	SelectControl,
+	PanelBody,
+	ToolbarGroup,
+	Dropdown,
+	Path,
+} from '@wordpress/components';
+import {
+	BlockControls,
+	InspectorControls,
+	RichText,
+	PanelColorSettings,
+} from '@wordpress/block-editor';
+import { DOWN } from '@wordpress/keycodes';
+
+/**
+ * Internal dependencies
+ */
+import { countryCodes } from '../shared/countrycodes.js';
+import renderMaterialIcon from '../../../shared/render-material-icon';
+import HelpMessage from '../../../shared/help-message';
+import './view.scss';
+
+const WHATSAPP_GREEN = '#25D366';
+const WHATSAPP_DARK = '#465B64';
+const WHATSAPP_LIGHT = '#F4F4F4';
+
+export default function WhatsAppButtonEdit( { attributes, setAttributes, className, clientId } ) {
+	const {
+		countryCode,
+		phoneNumber,
+		buttonText,
+		firstMessage,
+		colorClass,
+		backgroundColor,
+	} = attributes;
+
+	const [ isValidPhoneNumber, setIsValidPhoneNumber ] = useState( true );
+	const { selectBlock } = useDispatch( 'core/block-editor' );
+
+	const getCountryCode = useCallback( async () => {
+		setAttributes( { countryCode: '1' } );
+
+		const geoFetch = await fetch( 'http://ip-api.com/json/?fields=countryCode' )
```

Comment:
This call to a third party service raises a concern about privacy and possibly performance. Have we discussed / addressed those concerns?
Maybe It could be worth using our own geo API instead: https://public-api.wordpress.com/geo/
which we use in Jetpack here: https://github.com/Automattic/jetpack/commit/e9cd5c25362f5fcc5882489e637f8851d9423d52#diff-211293c591f97b51495cfc6047d441c6R224

Reply:
TIL!